app/src/main/java/org/simple/clinic/teleconsultlog/prescription/TeleconsultPrescriptionEffectHandler.kt

```diff
@@ -27,9 +37,30 @@ class TeleconsultPrescriptionEffectHandler @AssistedInject constructor(
         .addAction(GoBack::class.java, uiActions::goBackToPreviousScreen, schedulersProvider.ui())
         .addAction(ShowSignatureRequiredError::class.java, uiActions::showSignatureRequiredError, schedulersProvider.ui())
         .addTransformer(LoadDataForNextClick::class.java, loadDataForNextClick())
+        .addTransformer(CreatePrescription::class.java, createPrescription())
         .build()
   }
 
+  private fun createPrescription(): ObservableTransformer<CreatePrescription, TeleconsultPrescriptionEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulersProvider.io())
+          .doOnNext { medicalRegistrationIdPreference.set(Optional.of(it.medicalRegistrationId)) }
```

Comment:
We're mixing creating a prescription and updating the current medical registration ID in the same effect, which should not be done. We can send out multiple effects in a single `dispatch` or `next`, which we should be using here to save the medical registration ID locally.

Reply:
Okay yeah, since these are side effects and doesn't depend on each other they can be different effects right?

Reply:
Yes